### ISSUE TITLE
Does springfox-swagger2:2.1.2 work with Scala?

### ISSUE THREAD

**Issue by cer** [springfox/springfox#897]
I have a Scala/Spring Boot application.
JSon de/serialization is handled by an ObjectMapper configured with DefaultScalaModule along with JodaModule and some application specific customizations.
I defined an ApplicationListener for the ObjectMapperConfigured that configures the supplied ObjectMapper.

Springfox Swagger finds the controller methods but has an empty `{}` model schema for the requests and responses.

Suggestions?

**Comment by dilipkrish** [comment]
Frankly I've never played much scala + springboot combination. 

I don't think there is anything special about scala. It should just work fine. Would you be able to share a repo I can take a look at perhaps? If not would you mind sharing how you're configuring your `Docket`? If you've given it an explicit group name then you might be hitting the wrong endpoint.

**Comment by cer** [comment]
I updated one of public examples: wip-swagger-support branch https://github.com/cer/microservices-examples/commits/wip-swagger-integration

This is the commit showing the changes: https://github.com/cer/microservices-examples/commit/03e5c931758239fb69ce60b3bba3287f7e8e0d24

**Comment by dilipkrish** [comment]
@cer I dug into this some. It seems like we've run into un-anticipated usage territory :) So everything you've done is perfectly correct to set it up. 

The problem is the way scala sets these objects up. The expectation is that these are java beans in the library, but the scala objects these are coming through as parameters (not fields or accessor methods). Will add a chore to accommodate that exception. 

Thanks for reporting this issue! One way to get around this is for now until we have a fix out define the `RegistrationRequest` and `RegistrationResponse` as you would a java bean if thats possible (sorry Im not super familiar with scala.)

**Comment by dilipkrish** [comment]
Just to clarify interfering bean properties via constructors or via factory methods annotated with `@JsonCreator` isn't yet supported. I suspect it isn't hard to fix :)

**Comment by mkotsbak** [comment]
Does it work if you annotate the fields of RegistrationRequest, RegistrationResponse and NewRegistrationNotification with @BeanProperty?

**Comment by dilipkrish** [comment]
@mkotsbak it did not unfortunately. However it should be fixed in the latest 2.1.3-SNAPSHOT. @cer would you mind trying it with the latest snapshot? Plan on doing a release in the next few days.

**Comment by doctorstrange85** [comment]
There are problems with scala Option, List, Map, Set, e.t.c. 
Scala enumeration is a problem too. 
I've managed to fix quickly Option[_]

```scala
new Docket(DocumentationType.SWAGGER_2)
      .produces(Set("application/json").asJava)
      .select.apis(RequestHandlerSelectors.withClassAnnotation(classOf[RestController]))
      .paths(PathSelectors.any)
      .build
      .useDefaultResponseMessages(false)
      .genericModelSubstitutes(classOf[Option[_]])
      .directModelSubstitute(classOf[List[_]], classOf[java.util.List[_]])
      .directModelSubstitute(classOf[Set[_]], classOf[java.util.List[_]])
      .directModelSubstitute(classOf[Map[_, _]], classOf[java.util.Map[_, _]])
      .apiInfo(this.apiInfo).enable(this.properties.enabled)
```
`.genericModelSubstitutes(classOf[Option[_]])` - it works. Generic type is exposed in UI
None of `directModelSubstitute` worked.
scala List of Strings  presented as
```json
"serverAddresses": {
      "empty": true,
      "traversableAgain": true
    }
```
Scala enumerations look this way:


There is a swagger module for scala
`io.swagger:swagger-scala-module_2.12:1.0.6`  but I didn't find a way to register
`class SwaggerScalaModelConverter extends io.swagger.converter.ModelConverter`
at springfox...
